docs/next/en-US/state/revalidation.md

```diff
@@ -10,15 +10,15 @@ The first type of revalidation is the simplest: you set a schedule with `.revali
+Crucially, this is lazy revalidation: Perseus will not immediately revalidate a page once the revalidation interval is reached. For example, if our news site isn't very popular for its first month, and only gets two visits per day, it won't revalidate 24 times, it will probably revalidate twice: because only two people visited. This also means that revalidation can behave in unexpected ways. Let's say you have a page that revalidates every five seconds, and it's built at second 0. If, no-one requests it until second 6, and then there's a request every second, it will revalidate at second 6, then second 11, then second 16, etc. You may need to re-read that to understand this, and it's usually not a problem, unless you have very strict requirements.
 
-Note that this is all page-specific, so it's entirely possible for two different pages in the same template to have teh same revalidation interval and revalidate at different times.
+Note that this is all page-specific, so it's entirely possible for two different pages in the same template to have the same revalidation interval and revalidate at different times.
 
 ## Logic-based revalidation
 
-When you have more stringent needs, you might wish to use logic-based revalidation, which is based on the `.should_revalidate_fn()` method on `Template`. To this, you provide an `async` function of the usual sort with the usual `BlamedError<E>` error handling (see [here](:state/build) for an explanation of that) that takes a [`StateGeneratorInfo`](=prelude/struct.StateGeneratorInfo@perseus) instance and the user's request, and you return a `bool`: if it's true, the page will revalidate, but, if `false`, the old state will stand. This can be used to do more advanced things like having a database of new news, but also having a micro-site set to tell you whether or not there is new news. Thus, you can perform the quicker check to the micro-site (which acts as a [canary](https://en.wikipedia.org/wiki/Sentinel_species)) to avoid unnecessary revalidations, which will improve performance.
+When you have more stringent needs, you might wish to use logic-based revalidation, which is based on the `.should_revalidate_fn()` method on `Template`. To this, you provide an `async` function of the usual sort with the usual `BlamedError<E>` error handling (see [here](:state/build) for an explanation of that) that takes a [`StateGeneratorInfo`](=prelude/struct.StateGeneratorInfo@perseus) instance and the user's request, and you return a `bool`: if it's true, the page will revalidate, but, if `false`, the old state will stand. This can be used to do more advanced things like having a database of new news, but also having a micro-site set to tell you whether there is new news. Thus, you can perform the quicker check to the micro-site (which acts as a [canary](https://en.wikipedia.org/wiki/Sentinel_species)) to avoid unnecessary revalidations, which will improve performance.
```

Comment:
```suggestion
When you have more stringent needs, you might wish to use logic-based revalidation, which is based on the `.should_revalidate_fn()` method on `Template`. To this, you provide an `async` function of the usual sort with the usual `BlamedError<E>` error handling (see [here](:state/build) for an explanation of that) that takes a [`StateGeneratorInfo`](=prelude/struct.StateGeneratorInfo@perseus) instance and the user's request, and you return a `bool`: if it's true, the page will revalidate, but, if `false`, the old state will stand. This can be used to do more advanced things like having a database of new news, but also having a micro-site set to tell you whether or not there is new news. Thus, you can perform the quicker check to the micro-site (which acts as a [canary](https://en.wikipedia.org/wiki/Sentinel_species)) to avoid unnecessary revalidations, which will improve performance.
```

Again, whether or not.